docs/workflows/best-practices-architecture.md

```diff
@@ -0,0 +1,76 @@
+# Guidelines for designing a component architecture
+
+Software architecture has a central position when it comes to and managing the life cycle of large-scale software landscapes. A key part of a healthy software architecture is designing a healthy and future-proof component architecture. By incorporating functional and technical aspects in reusable blocks, component architectures allow for higher-level understanding of the software landscape as a whole. This has many benefits, but is in particular useful when considering system-wide concerns such as system evolution. However, with the ever-growing complexity and the move towards distributed and service-oriented software systems, this is becoming an increasingly challenging task to achieve.
+
+Tracking your architecture evolution is therefore key to ensuring your software landscapes and its systems remain future-proof. Every architecture is bound to change over time, due to a changing environment in which it operates. These guidelines are meant to help architects and development teams with their component architecture, based on best practices.
+
+<sig-toc></sig-toc>
+
+## Why modularize your codebase?
+
+Every non-trivial system is split into components. However, there are multiple different reasons for why people decide to modularize their system. Some of these reasons strictly relate to technical benefits, and other reasons purely relate to making life easier for the people working on the system. 
```

Comment:
"However, there are multiple different reasons for why people decide to modularize their system. Some of these reasons strictly relate to technical benefits, and other reasons purely relate to making life easier for the people working on the system." --> make it shorter and more prescriptive, e.g.: "There are multiple reasons for modularizing your software system, some relate to technical benefits, some to  improve the way of working and co-operation of the developers."